core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -23,377 +22,271 @@ import scala.jdk.CollectionConverters._
-        )
+  def getSpendExportConfiguration(project: RawlsBillingProjectName): Future[BillingProjectSpendExport] = dataSource
+    .inTransaction(_.rawlsBillingProjectQuery.getBillingProjectSpendConfiguration(project))
+    .recover { case _: RawlsException =>
+      throw RawlsExceptionWithErrorReport(
+        StatusCodes.BadRequest,
+        s"billing account not found on billing project ${project.value}"
       )
-      SpendReportingForDateRange(
-        cost.toString(),
-        credits.toString(),
-        currency.getCurrencyCode,
-        workspace = Option(workspaceName),
-        googleProjectId = Option(googleProjectId),
-        subAggregation = subAggregation
+    }
+    .map {
+      _.getOrElse(
+        throw RawlsExceptionWithErrorReport(StatusCodes.NotFound, s"billing project ${project.value} not found")
       )
-    }.toList
-
-    SpendReportingAggregation(
-      SpendReportingAggregationKeys.Workspace,
-      workspaceSpend
-    )
-  }
+    }
 
-  private def extractCategorySpendAggregation(rows: List[FieldValueList],
-                                              currency: Currency,
-                                              subAggregationKey: Option[SpendReportingAggregationKey] = None,
-                                              workspaceProjectsToNames: Map[GoogleProject, WorkspaceName] = Map.empty
-  ): SpendReportingAggregation = {
-    val spendByCategory: Map[TerraSpendCategory, List[FieldValueList]] =
-      rows.groupBy(row => TerraSpendCategories.categorize(row.get("service").getStringValue))
-    val categorySpend = spendByCategory.map { case (category, rowsForCategory) =>
-      val (cost, credits) = sumCostsAndCredits(rowsForCategory, currency)
-      val subAggregation = subAggregationKey.map { key =>
-        extractSpendAggregation(rowsForCategory,
-                                currency,
-                                aggregationKey = SpendReportingAggregationKeyWithSub(key),
-                                workspaceProjectsToNames = workspaceProjectsToNames
-        )
-      }
-      SpendReportingForDateRange(
-        cost.toString,
-        credits.toString,
-        currency.getCurrencyCode,
-        category = Option(category),
-        subAggregation = subAggregation
-      )
-    }.toList
+  def getWorkspaceGoogleProjects(projectName: RawlsBillingProjectName): Future[Map[String, WorkspaceName]] =
+    dataSource.inTransaction(_.workspaceQuery.listWithBillingProject(projectName)).map {
+      _.collect {
+        case w if w.workspaceVersion == WorkspaceVersions.V2 => w.googleProjectId.value -> w.toWorkspaceName
+      }.toMap
+    }
 
-    SpendReportingAggregation(
-      SpendReportingAggregationKeys.Category,
-      categorySpend
+  def validateReportParameters(startDate: DateTime, endDate: DateTime): Unit = if (startDate.isAfter(endDate)) {
+    throw RawlsExceptionWithErrorReport(
```

Comment:
Out of scope as well but: We should think about pursuing more specific exception types for this class as well, rather than reusing RawlsExceptionWithErrorReport everywhere.